mmv1/third_party/terraform/services/secretmanager/data_source_secret_manager_secret_version.go

```diff
@@ -147,11 +152,21 @@ func dataSourceSecretManagerSecretVersionRead(d *schema.ResourceData, meta inter
 	}
 
 	data := resp["payload"].(map[string]interface{})
-	secretData, err := base64.StdEncoding.DecodeString(data["data"].(string))
-	if err != nil {
-		return fmt.Errorf("Error decoding secret manager secret version data: %s", err.Error())
+	var secretData string
+	dIsSecretDataBase64, ok := d.Get("is_secret_data_base64").(bool)
```

Comment:
nit: You can omit the `ok`- the return of d.Get is guaranteed to be a `bool` by the schema, as the type is TypeBool.

Reply:
Yeah, I got your point and I've removed the checks as discussed. However, I originally included them to enhance the defensiveness of the type assertions based on the review comments I received on PR #11754.

Out of curiosity, are there specific scenarios—perhaps during string type assertions—where it would be advisable to retain such checks?